emission/tests/modellingTests/TestRunGreedyIncrementalModel.py

```diff
@@ -0,0 +1,186 @@
+        self.initial_similar_trips = len(self.similar_trips)
+        self.expected_trips = self.initial_similar_trips + self.new_trips_per_invocation
+        logging.debug(f"end of test, expecting {self.expected_trips} trips")
+
+        # find the centroid of the similar trip data
+        src_x, src_y, dst_x, dst_y = np.mean(self.similar_features, axis=0)
+        self.origin = [src_x, src_y]
+        self.destination = [dst_x, dst_y]
+
+
+    def tearDown(self):
+        """
+        delete entries for user self.user_id in the database, not
+        yet implemented in database operations, so these test entries will
+        have to stick around for now.
+        """
+        pass
+
+    def testIncrementalRun(self):
+        """
+        incremental trip models train from Confirmedtrip entries at most
+        once. to test this behavior, a model is built based on a small
+        Confirmedtrip dataset stored at a file location (See setUp, above).
+        this happens once and is not repeated when the test is re-run,
+        unless a new database instance is spun up. within the test method,
+        an additional few mock trips are generated with a later timestamp.
+        the training model should 1) only see the new trips, 2) have been
+        trained on the expected number of trips at completion.
+        """
+        # create a new trip sampling from the centroid and the existing
+        # set of user input data
+        # timestamps for these rows cannot be within the last 5 seconds
+        # based on invariant set in pipeline_queries.py by the
+        # END_FUZZ_AVOID_LTE constant.
+        label_data = etmm.extract_trip_labels(self.similar_trips)
+        new_trips = etmm.generate_mock_trips(
+            user_id=self.user_id,
+            trips=self.new_trips_per_invocation,
+            origin=self.origin,
+            destination=self.destination,
+            label_data=label_data,
+            threshold=0.0001, # ~10m,
+            start_ts=time.time() - 20,
+            end_ts=time.time() - 10
+        )
+
+        self.ts.bulk_insert(new_trips)
+        all_trips = list(self.ts.find_entries([esdatq.CONFIRMED_TRIP_KEY]))
+        logging.debug(f'total of {len(all_trips)} now stored in database')
+
+        # train the new model on the complete collection of trips
+        eamur.update_trip_model(
+            user_id=self.user_id,
+            model_type=self.model_type,
+            model_storage=self.model_storage,
+            min_trips=self.initial_similar_trips,
+            model_config=self.greedy_model_config
+        )
+        updated_model = eamur._load_stored_trip_model(
+            self.user_id,
+            model_type=self.model_type,
+            model_storage=self.model_storage,
+            model_config=self.greedy_model_config
+        )
+
+        self.assertEqual(len(updated_model.bins), 2, 
+            'there should be two bins, one with similar trips, one with an outlier')
+
```

Comment:
Where the outlier is the 6th trip from the setup, right? Can you please add a comment here or in the following assert to clarify?